File: src/main/java/com/comma/newsbot/formatter/MessageFormatter.java
package com.comma.newsbot.formatter;

import com.comma.newsbot.domain.News;
import org.springframework.stereotype.Component;

@Component
public class MessageFormatter {
    public String formatNewsToMessage(News news){

        final int MAX_DESCRIPTION_LENGTH = 1000;

        String rawDescription = (news.getDescription() != null && !news.getDescription().isEmpty())
                ? news.getDescription()
                : "Check the link for more details!";

        String safeDescription;
        if (rawDescription.length() > MAX_DESCRIPTION_LENGTH) {
            safeDescription = rawDescription.substring(0, MAX_DESCRIPTION_LENGTH) + "... (read more)";
        } else {
            safeDescription = rawDescription;
        }


        return """
               ☕ **Time for a Comma ,**
               
               Life moves fast. Take a break and check this out:
               
               📰 **%s**
               
               > %s
               
               🔗 [Read full story](%s)
               """.formatted(
                news.getTitle(),
                safeDescription,
                news.getUrl()
        );
    }
}

File: src/formatter/message_formatter.py
from src.domain.news import News

MAX_DESCRIPTION_LENGTH = 1500


def format_news_to_message(news: News) -> str:
    raw = news.description if news.description else "Check the link for more details!"
    description = (
        raw[:MAX_DESCRIPTION_LENGTH] + "... (read more)"
        if len(raw) > MAX_DESCRIPTION_LENGTH
        else raw
    )
    return (
        "☕ **Time for a Comma ,**\n\n"
        "Life moves fast. Take a break and check this out:\n\n"
        f"📰 **{news.title}**\n\n"
        f"> {description}\n\n"
        f"🔗 [Read full story]({news.url})"
    )
